src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -48,15 +75,402 @@ pub fn make_distro() -> PackageDistro {
     }
 }
 
-fn package_items_names(distro: &PackageDistro) -> Vec<String> {
-    distro
-        .services
+pub fn make_distro_default() -> PackageDistro {
+    let decider_settings = DeciderConfig {
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: "http://127.0.0.1:12009".to_string(),
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0x0".to_string(),
+    };
+    // let's try to run a decider cycle on demand by updating the config
+    let mut trigger_config = TriggerConfig::default();
+    trigger_config.clock.start_sec = 1;
+    make_distro(trigger_config, decider_settings)
+}
+
+pub fn make_distro_with_api(api: String) -> PackageDistro {
+    let decider_settings = DeciderConfig {
+        // worker will run once
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: api,
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f"
+            .to_string(),
+    };
+    // decider will run once
+    let mut trigger_config = TriggerConfig::default();
+    make_distro(trigger_config, decider_settings)
+}
+
+pub fn make_distro_with_api_and_config(api: String, config: TriggerConfig) -> PackageDistro {
+    let decider_settings = DeciderConfig {
+        // worker will run once
+        worker_period_sec: 0,
+        worker_ipfs_multiaddr: "/ip4/127.0.0.1/tcp/5001".to_string(),
+        chain_api_endpoint: api,
+        chain_network_id: 11,
+        chain_contract_block_hex: "0x0".to_string(),
+        chain_matcher_addr: "0x0".to_string(),
+        chain_workers_gas: 210_00,
+        chain_wallet_key: "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f"
+            .to_string(),
+    };
+    // decider will run once
+    make_distro(config, decider_settings)
+}
+
+async fn execute(
```

Comment:
WDYT, should that be here or in nox's test utils?

I'm thinking about whether this and some other functions from here would be useful in tests in other spells, eg in fluencelabs/spell.

Reply:
would be nice, not sure if exactly this interface is good for a lib though